src/services/polling.service.ts

```diff
@@ -0,0 +1,201 @@
+        TransactionSubStatus[tr.subStatus as keyof typeof TransactionSubStatus],
+      destinations: tr.destinations?.map((dest) => ({
+        ...dest,
+        amountUSD: Number(dest.amountUSD),
+      })),
+      signedMessages: tr.signedMessages!,
+    };
+  }
+
+  private async pollAndUpdateAll() {
+    console.log("PollingService: pollAndUpdateAll");
+    try {
+      // Get transactions from Fireblocks:
+      const txsResponses: TransactionResponse[] = await fetchAllTxs(
+        (pagePath) =>
+          this.clients.admin.getTransactionsWithPageInfo({}, pagePath),
+      );
+      for (const txResponse of txsResponses) {
+        const tx = this.txResponseToTxDetails(txResponse);
+        const { id, status } = tx;
+        await patchTransactionAmountUsd(tx, this.clients.cmc);
+        await handleTransactionCreatedOrUpdated(id, status, tx);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  /**
+   * Increases the current polling frequency for a short period of time
+   */
+  private increasePollingFrequency() {
+    console.log(
+      "PollingService: increasePollingFrequency - increase frequency",
+    );
+    this.currentPollingIntervalMs = this.pollingShortIntervalMs;
+
+    if (this.incomingTxTimer) {
+      clearTimeout(this.incomingTxTimer);
+      this.incomingTxTimer = null;
+    }
+
+    this.incomingTxTimer = setTimeout(() => {
+      console.log(
+        "PollingService: increasePollingFrequency - reset back to normal",
+      );
+      this.currentPollingIntervalMs = this.pollingIntervalMs;
+    }, 120_000);
+  }
+
+  /**
+   * Polls for recent transactions from FB BE and updates their status in DB.
+   */
+  async pollAndUpdate() {
+    try {
+      // Get recent transactions from DB (ignore transactions that were not updated for more than 48 hours):
+      const dbRecentTxs = await this.getDbRecentTransactionsSorted(
+        this.hoursAgoDate(48),
+      );
+      // Get createdAt of earliest non-completed tx
+      const firstNonCompletedTx = dbRecentTxs.find(
+        (tx) => !this.isCompletedTx(tx),
+      );
+      const minCreatedAt: Date =
+        firstNonCompletedTx?.createdAt || this.hoursAgoDate(48);
+      // Get transactions from Fireblocks:
+      const txsResponses: TransactionResponse[] = await fetchAllTxs(
+        (pagePath) =>
+          this.clients.admin.getTransactionsWithPageInfo(
+            { after: minCreatedAt.getTime() - 1 },
+            pagePath,
+          ),
+      );
+
+      const dbRecentTxsById = groupBy(dbRecentTxs, (tx) => tx.id);
+      for (const txResponse of txsResponses) {
+        const tx = this.txResponseToTxDetails(txResponse);
+        const { id, status } = tx;
+        await patchTransactionAmountUsd(tx, this.clients.cmc);
+
+        try {
+          if (!dbRecentTxsById[tx.id]) {
+            // Tx doesn't exist in DB yet
+            await handleTransactionCreated(id, status, tx);
+            this.increasePollingFrequency();
```

Comment:
this is because we expect the tx's status to update in the near time, right?
instead of increasing the polling for all, WDYT about doing so just for that tx?

Reply:
indeed it can be done,
but It will further complicate the code and I'm not sure it worths it.
WDYT?

Reply:
done